libuavcan/include/libuavcan/platform/memory.hpp

```diff
@@ -304,6 +323,126 @@ class LIBUAVCAN_EXPORT PoolAllocator
     }
 };
 
+namespace
```

Comment:
While working on the PyUAVCAN serialization logic I noticed that the complexity of the bit copy algorithm (in terms of operations per byte copied) can be reduced by taking advantage of the fact that either the source or the destination are always aligned at one byte:

- https://github.com/UAVCAN/pyuavcan/blob/uavcan-v1.0/pyuavcan/dsdl/_serialized_representation/_serializer.py#L162-L172
- https://github.com/UAVCAN/pyuavcan/blob/uavcan-v1.0/pyuavcan/dsdl/_serialized_representation/_deserializer.py#L221-L252

More specifically, for N bytes to copy, an implementation that assumes that either the source or the destination are aligned performs N or N+1 iterations of the outer loop (depending on the alignment at the opposite end), compared to (approximating here) N...2N of the generic method.

Reply:
Okay. Yeah. splitting this up into to methods simplifies the logic quite a bit. I'll also just hard-code uint8_t as the work type to make this endian agnostic.